pkg/counter/cache.go

```diff
@@ -55,42 +58,56 @@ func Reset() {
 
 func reset() {
 	cachedCounts.Counts = Counts{
-		UserAccountsPerClusterCounts: map[string]int{},
-		UsersPerActivationCounts:     map[string]int{},
+		UserAccountsPerClusterCounts:    map[string]int{},
+		UsersPerActivationCounts:        map[string]int{},
+		MasterUserRecordPerDomainCounts: map[string]int{},
 	}
 	cachedCounts.initialized = false
-	metrics.MasterUserRecordGauge.Set(float64(0))
 	metrics.UserAccountGaugeVec.Reset()
 	metrics.UsersPerActivationGaugeVec.Reset()
+	metrics.MasterUserRecordGauge.Set(float64(0))
+	metrics.MasterUserRecordGaugeVec.Reset()
 }
 
 // IncrementMasterUserRecordCount increments the number of MasterUserRecord in the cached counter
-func IncrementMasterUserRecordCount() {
+func IncrementMasterUserRecordCount(domain metrics.Domain) {
 	write(func() {
 		cachedCounts.MasterUserRecordCount++
 		log.Info("incrementing MasterUserRecordGauge", "value", cachedCounts.MasterUserRecordCount)
 		metrics.MasterUserRecordGauge.Set(float64(cachedCounts.MasterUserRecordCount))
+
+		cachedCounts.MasterUserRecordPerDomainCounts[string(domain)]++
+		log.Info("incrementing MasterUserRecordGaugeVec", "value", cachedCounts.MasterUserRecordPerDomainCounts[string(domain)], "domain", domain)
+		metrics.MasterUserRecordGaugeVec.WithLabelValues(string(domain)).Set(float64(cachedCounts.MasterUserRecordPerDomainCounts[string(domain)]))
 	})
 }
 
 // DecrementMasterUserRecordCount decreases the number of MasterUserRecord in the cached counter
-func DecrementMasterUserRecordCount(logger logr.Logger) {
+func DecrementMasterUserRecordCount(domain metrics.Domain) {
```

Comment:
Is there a reason to remove the logger passed into the function? I think it's passed in so that we know which controller was calling the function?

The `DecrementUserAccountCount` function still has logger as a param, so whatever we decide, we should probably make them consistent.

Reply:
yes, that's a fair point. I'll restore that

Reply:
restored in 0fd0a08